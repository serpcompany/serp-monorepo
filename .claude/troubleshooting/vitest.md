https://vitest.dev/guide/common-errors.html