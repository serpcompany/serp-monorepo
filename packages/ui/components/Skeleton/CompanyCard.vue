<template>
  <UCard
    :ui="{
      root: 'flex flex-col divide-dashed transition-all',
      body: 'p-5 sm:p-5 flex-grow',
      footer: `px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg`,
    }"
    variant="outline"
  >
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)">
          <USkeleton class="h-12 w-12 rounded-sm" />
        </div>
        <div class="flex flex-1 items-center gap-x-2">
          <div class="flex flex-col gap-2">
            <USkeleton class="h-7 w-32" />
          </div>
        </div>
      </div>
      <div class="flex flex-col gap-6">
        <USkeleton class="h-3.5" />
        <div class="inline-flex flex-wrap items-center gap-2">
          <USkeleton
            v-for="index in 3"
            :key="index"
            class="h-4 w-16 rounded-full"
          />
        </div>
      </div>
    </div>
    <template #footer>
      <div class="flex flex-row items-center gap-3">
        <USkeleton class="h-7 w-20" />
        <USkeleton class="ml-auto h-7 w-7" />
      </div>
    </template>
  </UCard>
</template>
