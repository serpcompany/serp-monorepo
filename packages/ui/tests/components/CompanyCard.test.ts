import { describe, expect, it } from 'vitest';
import CompanyCard from '../../components/CompanyCard.vue';
import ComponentRender from '../componentRender';
import '../mockUseUserSession';

let config_: Record<string, unknown> = {
  app: { baseURL: '/' },
  public: {
    useAuth: true,
  },
};

describe('companyCard Snapshot', () => {
  it.each([
    [
      'renders non-featured company correctly (without auth)',
      {
        props: {
          company: {
            id: 1,
            slug: 'non-featured',
            name: 'Non Featured Co.',
            featured: false,
            oneLiner: 'Just a regular company.',
            serplyLink: 'https://nonfeatured.example.com',
            upvotes: [],
          },
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: false,
          },
        },
      },
    ],
    [
      'renders featured company with complete details correctly (without auth)',
      {
        props: {
          company: {
            id: 2,
            slug: 'featured',
            name: 'Featured Co.',
            featured: true,
            oneLiner: 'A standout company.',
            excerpt: 'This company is truly awesome and innovative.',
            rating: 4.5,
            logo: 'https://featured.example.com/logo.png',
            screenshots: ['https://featured.example.com/screenshot.png'],
            features: [
              { id: 1, item: 'Innovation', description: 'Cutting edge tech.' },
              { id: 2, item: 'Quality', description: 'High quality products.' },
            ],
            serplyLink: 'https://featured.example.com',
            upvotes: ['user@example.com'],
          },
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: false,
          },
        },
      },
    ],
    [
      'renders company with expanded content correctly (without auth)',
      {
        props: {
          company: {
            id: 3,
            slug: 'expanded',
            name: 'Expanded Co.',
            featured: true,
            oneLiner: 'Company with extra details.',
            excerpt: 'Expanded content goes here.',
            rating: 3.8,
            // No logo provided; should fall back to screenshot
            screenshots: ['https://expanded.example.com/screenshot.png'],
            features: [
              {
                id: 3,
                item: 'Flexibility',
                description: 'Adaptable solutions.',
              },
            ],
            serplyLink: 'https://expanded.example.com',
            upvotes: [],
          },
          showReadMore: true,
          showFeatures: true,
          showExpandedContent: true,
          baseSlug: 'products/',
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: false,
          },
        },
      },
    ],
    [
      'renders non-featured company correctly (with auth)',
      {
        props: {
          company: {
            id: 1,
            slug: 'non-featured',
            name: 'Non Featured Co.',
            featured: false,
            oneLiner: 'Just a regular company.',
            serplyLink: 'https://nonfeatured.example.com',
            upvotes: [],
          },
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: true,
          },
        },
      },
    ],
    [
      'renders featured company with complete details correctly (with auth)',
      {
        props: {
          company: {
            id: 2,
            slug: 'featured',
            name: 'Featured Co.',
            featured: true,
            oneLiner: 'A standout company.',
            excerpt: 'This company is truly awesome and innovative.',
            rating: 4.5,
            logo: 'https://featured.example.com/logo.png',
            screenshots: ['https://featured.example.com/screenshot.png'],
            features: [
              { id: 1, item: 'Innovation', description: 'Cutting edge tech.' },
              { id: 2, item: 'Quality', description: 'High quality products.' },
            ],
            serplyLink: 'https://featured.example.com',
            upvotes: ['user@example.com'],
          },
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: true,
          },
        },
      },
    ],
    [
      'renders company with expanded content correctly (with auth)',
      {
        props: {
          company: {
            id: 3,
            slug: 'expanded',
            name: 'Expanded Co.',
            featured: true,
            oneLiner: 'Company with extra details.',
            excerpt: 'Expanded content goes here.',
            rating: 3.8,
            // No logo provided; should fall back to screenshot
            screenshots: ['https://expanded.example.com/screenshot.png'],
            features: [
              {
                id: 3,
                item: 'Flexibility',
                description: 'Adaptable solutions.',
              },
            ],
            serplyLink: 'https://expanded.example.com',
            upvotes: [],
          },
          showReadMore: true,
          showFeatures: true,
          showExpandedContent: true,
          baseSlug: 'products/',
        },
        config: {
          app: { baseURL: '/' },
          public: {
            useAuth: true,
          },
        },
      },
    ],
  ])(
    '%s',
    async (desc: string, options: { props: unknown; config?: unknown }) => {
      config_ = options.config;
      mockNuxtImport('useRuntimeConfig', () => () => config_);
      const html = await ComponentRender(
        `CompanyCard ${desc}`,
        options,
        CompanyCard,
      );
      expect(html).toMatchSnapshot();
    },
  );
});
