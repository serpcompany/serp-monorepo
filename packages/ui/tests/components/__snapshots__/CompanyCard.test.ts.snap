// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyCard Snapshot > renders company with expanded content correctly (with auth) 1`] = `
"<div class="rounded-lg bg-default divide-y divide-default flex flex-col divide-dashed transition-all ring-secondary-400 hover:ring-secondary-300 dark:ring-secondary-600 dark:hover:ring-secondary-500 ring-2 lg:col-span-2">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><img width="48" height="48" data-nuxt-img="" srcset="https://expanded.example.com/screenshot.png 1x, https://expanded.example.com/screenshot.png 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://expanded.example.com/screenshot.png"></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/expanded/reviews/" class="transition-all hover:underline">Expanded Co.</a></h2><a href="https://expanded.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm">Expanded content goes here.</p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://expanded.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;

exports[`CompanyCard Snapshot > renders company with expanded content correctly (without auth) 1`] = `
"<div class="rounded-lg bg-default divide-y divide-default flex flex-col divide-dashed transition-all ring-secondary-400 hover:ring-secondary-300 dark:ring-secondary-600 dark:hover:ring-secondary-500 ring-2 lg:col-span-2">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><img width="48" height="48" data-nuxt-img="" srcset="https://expanded.example.com/screenshot.png 1x, https://expanded.example.com/screenshot.png 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://expanded.example.com/screenshot.png"></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/expanded/reviews/" class="transition-all hover:underline">Expanded Co.</a></h2><a href="https://expanded.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm">Expanded content goes here.</p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://expanded.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;

exports[`CompanyCard Snapshot > renders featured company with complete details correctly (with auth) 1`] = `
"<div class="rounded-lg bg-default divide-y divide-default flex flex-col divide-dashed transition-all ring-secondary-400 hover:ring-secondary-300 dark:ring-secondary-600 dark:hover:ring-secondary-500 ring-2 lg:col-span-2">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><img width="48" height="48" data-nuxt-img="" srcset="https://featured.example.com/logo.png 1x, https://featured.example.com/logo.png 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://featured.example.com/logo.png"></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/featured/reviews/" class="transition-all hover:underline">Featured Co.</a></h2><a href="https://featured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm">This company is truly awesome and innovative.</p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://featured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;

exports[`CompanyCard Snapshot > renders featured company with complete details correctly (without auth) 1`] = `
"<div class="rounded-lg bg-default divide-y divide-default flex flex-col divide-dashed transition-all ring-secondary-400 hover:ring-secondary-300 dark:ring-secondary-600 dark:hover:ring-secondary-500 ring-2 lg:col-span-2">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><img width="48" height="48" data-nuxt-img="" srcset="https://featured.example.com/logo.png 1x, https://featured.example.com/logo.png 2x" role="img" class="h-full w-full rounded-[inherit] object-cover" src="https://featured.example.com/logo.png"></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/featured/reviews/" class="transition-all hover:underline">Featured Co.</a></h2><a href="https://featured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm">This company is truly awesome and innovative.</p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://featured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;

exports[`CompanyCard Snapshot > renders non-featured company correctly (with auth) 1`] = `
"<div class="rounded-lg bg-default ring ring-default divide-y divide-default flex flex-col divide-dashed transition-all hover:ring-(--ui-border-accented)">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><span class="iconify i-lucide:asterisk text-muted shrink-0" aria-hidden="true"></span></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/non-featured/reviews/" class="transition-all hover:underline">Non Featured Co.</a></h2><a href="https://nonfeatured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm"></p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://nonfeatured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;

exports[`CompanyCard Snapshot > renders non-featured company correctly (without auth) 1`] = `
"<div class="rounded-lg bg-default ring ring-default divide-y divide-default flex flex-col divide-dashed transition-all hover:ring-(--ui-border-accented)">
  <!--v-if-->
  <div class="p-5 sm:p-5 flex-grow">
    <div class="flex flex-col gap-4">
      <div class="flex flex-row gap-4">
        <div class="bg-muted size-auto rounded-sm border border-(--ui-border)"><span class="inline-flex items-center justify-center shrink-0 select-none overflow-hidden align-middle bg-elevated size-12 text-2xl rounded-sm"><span class="iconify i-lucide:asterisk text-muted shrink-0" aria-hidden="true"></span></span></div>
        <div class="flex flex-1 items-center gap-x-2">
          <h2 class="text-dark text-xl font-semibold"><a href="/products/non-featured/reviews/" class="transition-all hover:underline">Non Featured Co.</a></h2><a href="https://nonfeatured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted ml-auto">
            <!--v-if--><span class="truncate">Website</span><span class="iconify i-lucide:external-link shrink-0 size-4" aria-hidden="true"></span>
          </a>
        </div>
      </div>
      <div class="flex flex-col gap-4">
        <p class="text-muted text-sm"></p>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div class="p-4 px-5 py-4 sm:px-5 bg-gray-50 dark:bg-black/10 rounded-b-lg">
    <div class="flex flex-row items-center gap-3">
      <div class="relative inline-flex -space-x-px" module="company"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] gap-1 rounded-sm ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center text-sm"><!--v-if--><span class="truncate">NaN</span>
        <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-4" aria-hidden="true"></span>
          <!--v-if-->
          <!--v-if-->
        </button>
      </div><a href="https://nonfeatured.example.com" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors text-xs gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5 ml-auto"><span class="iconify i-lucide:bookmark shrink-0 size-4" aria-hidden="true"></span>
        <!--v-if-->
        <!--v-if-->
      </a>
    </div>
  </div>
</div>"
`;
